src/kubernetes_multus.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm Library used to leverage the Multus Kubernetes CNI in charms.
+
+- On charm installation, it will:
+  - Create the requested network attachment definitions
+  - Patch the statefultset with the necessary annotations for the container to have interfaces
+    that use those new network attachments.
+- On charm removal, it will:
+  - Delete the created network attachment definitions
+
+## Usage
+
+```python
+
+from kubernetes_multus import (
+    KubernetesMultusCharmLib,
+    NetworkAttachmentDefinition,
+    NetworkAnnotation
+)
+
+class YourCharm(CharmBase):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+        self._kubernetes_multus = KubernetesMultusCharmLib(
+            charm=self,
+            network_attachment_definitions=[
+                NetworkAttachmentDefinition(name="access-net"),
+                NetworkAttachmentDefinition(name="core-net"),
+            ],
+            network_annotations=[
+                NetworkAnnotation(name="access-net", interface="eth0"),
+                NetworkAnnotation(name="core-net", interface="eth1"),
+            ]
+        )
+```
+"""
+
+import json
+import logging
+from dataclasses import asdict, dataclass, field
+from typing import List, Optional, Union
```

Comment:
```suggestion
from typing import Optional, Union
```
This is not required with Python 3.10, the default version on 22.04

Reply:
Fixed